### PR TITLE
Refactor `tqec.circuit.schedule` into submodules

### LINKED ISSUE #358: Refactor `ScheduledCircuit` and implement a frozen alternative.
**Is your feature request related to a problem? Please describe.**
The current `schedule.py` file is quite huge (more than 900 lines) and contains a lot of code. It would be nice to split it into several smaller files, just like what was done with `BlockGraph` in #340.

At the same time, it would be nice to be able to ensure that some `ScheduledCircuit` instances will never be mutated. This will allow us to safely implement `__hash__` for `Plaquette` instances for example, and would avoid any kind of bugs due to unwanted in-place changes.

**Describe the solution you'd like**
Split `tqec.circuit.schedule` into submodules, e.g.:

- `tqec.circuit.schedule.schedule`
- `tqec.circuit.schedule.circuit`
- `tqec.circuit.schedule.frozen`
- `tqec.circuit.schedule.manipulation`

### PR DESCRIPTION
**Pull request by nelimee**
This PR fixes #358.

The first commit of this PR splits `src/tqec/circuit/schedule.py` into multiple files in the newly created folder `src/tqec/circuit/schedule/` and re-export all the public symbols in its `__init__.py`.

The subsequent commits will likely implement a `FrozenScheduledCircuit` data structure.

### CODE REVIEW

**Comment by nelimee** [comment]
Right now I am not sure that an implementation of `FrozenScheduledCircuit` is needed. I think this can be merged as is, and another PR might be opened for `FrozenScheduledCircuit` if we really need it.

File: src/tqec/circuit/schedule/manipulation.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
+"""Defines the :class:`ScheduledCircuit` class.
+
+This module is central to the `tqec` library because it implements one of the
+most used class of the library: :class:`ScheduledCircuit`.
+
+A :class:`ScheduledCircuit` is "simply" a `stim.Circuit` that have all its
+moments (portions of computation between two `TICK` instructions) assigned to
+a unique positive integer representing the time slice at which the moment should
+be scheduled.
```
Comment (inmzhang) [inline src/tqec/circuit/schedule/manipulation.py:9]: Seems in the wrong place.

File: src/tqec/circuit/schedule/exception.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from tqec.exceptions import TQECException
+
+
+class ScheduleException(TQECException):
+    pass
```
[the remark this answers is not in the document]
Comment (nelimee) [reply src/tqec/circuit/schedule/exception.py:5]: Depends on which direction we want #262 to evolve.

[the remark this answers is not in the document]
Comment (nelimee) [reply src/tqec/circuit/schedule/exception.py:5]: I think this can be kept here and might be removed (or improved) by #262.

**Review by inmzhang — APPROVED** [review]

**Pull request merged**